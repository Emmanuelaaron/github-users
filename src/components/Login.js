import { useSelector } from 'react-redux';
import UserForm from './UserForm';
import { LoginUser } from '../redux/session/Session';

const Login = () => {
  const currentState = useSelector((state) => state.session);
  return (
    (!currentState.session)
      ? <UserForm action="Login" userActionMethod={LoginUser} />
      : <h1>data here. yo logged in</h1>
  );
};
export default Login;
