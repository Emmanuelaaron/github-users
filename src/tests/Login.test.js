import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import store from '../redux/ConfigureStore';
import Login from '../components/Login';

describe('Login Component', () => {
  describe('use jest snapshot to test login component', () => {
    it('render Login compenent as expected', async () => {
      const loginCompenent = render(
        <Provider store={store}>
          <Login />
        </Provider>,
      );
      expect(await screen.findAllByText('Login here Please')).toBeTruthy();
      expect(loginCompenent.asFragment()).toMatchSnapshot();
    });
  });
});
