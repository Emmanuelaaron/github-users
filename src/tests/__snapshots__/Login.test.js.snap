// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component use jest snapshot to test login component render Login compenent as expected 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column align-items-center justify-content-center mt-5 pt-5 "
  >
    <h1>
      Login here Please
    </h1>
    <form
      class="w-50"
    >
      <div
        class="mb-3"
      >
        <div
          class="d-flex flex-column"
        >
          <label
            class="form-label"
            for="formBasicEmail"
          >
            Username
          </label>
          <label
            class="text-danger form-label"
            for="formBasicEmail"
          />
        </div>
        <input
          class="form-control"
          id="formBasicEmail"
          placeholder="Enter username"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="formBasicPassword"
        >
          Password
        </label>
        <input
          class="form-control"
          id="formBasicPassword"
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <div
        class="d-flex justify-content-between"
      >
        <button
          class="btn btn-primary"
          type="submit"
        >
          Login
        </button>
        <a
          class="nav-link"
          href="/signup"
        >
          SignUP
        </a>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
